src/structures/ReactionCollector.js

```diff
@@ -41,18 +41,35 @@ class ReactionCollector extends Collector {
 
     this.client.setMaxListeners(this.client.getMaxListeners() + 1);
     this.client.on('messageReactionAdd', this.listener);
+    this.client.on('messageReactionRemove', this.removeListener);
   }
 
   /**
    * Handle an incoming reaction for possible collection.
    * @param {MessageReaction} reaction The reaction to possibly collect
+   * @param {User} user The user that added the reaction
    * @returns {?{key: Snowflake, value: MessageReaction}}
    * @private
    */
-  handle(reaction) {
+  handle(reaction, user) {
     if (reaction.message.id !== this.message.id) return null;
     return {
-      key: reaction.emoji.id || reaction.emoji.name,
+      key: `${user.id}-${reaction.emoji.id || reaction.emoji.name}`,
```

Comment:
I'm not so sure about this, since I thought the intent of a reaction collector was to collect all of the reactions by emoji. There's no real benefit to keying it by the user ID as well, since multiple keys will now be pointing to the same `MessageReaction` value. For example:

1. User A reacts with 🤔
2. User B reacts with 🤔
3. User A reacts with <:POGGERS:440658615880253440>
4. User C reacts with <:POGGERS:440658615880253440>

There will now be 4 collected entries pointing to only 2 values. `User A-🤔` and `User B-🤔` will both point to a `MessageReaction` that has User A and B in its `users` collection; likewise with `User A-POGGERS` and `User C-POGGERS`.